packages/noir-ethereum-api/src/noir/oracles/accountOracles.ts

```diff
@@ -19,19 +20,44 @@ export interface AccountWithProof {
   depth: string;
 }
 
-export function serializeAccountWithProof(account: AccountWithProof): ForeignCallOutput[] {
-  return [account.balance, account.codeHash, account.nonce, account.key, account.value, account.proof, account.depth];
-}
-
 export const getAccountOracle = async (client: PublicClient, args: string[][]): Promise<ForeignCallOutput[]> => {
+  const { blockNumber, address } = parseNoirGetAccountArguments(args);
+  const accountProof: GetProofReturnType = await getAccountProof(client, address, blockNumber);
+  const accountWithProof = encodeAccount(accountProof);
+  return serializeAccountWithProof(accountWithProof);
```

Comment:
Why suddenly two methods `encodeAccount` and `serializeAccountWithProof`, we were following a pattern 
`encodeX(x: X): string[]`, why suddenly break it into two functions?

Reply:
ok, combined in one `encode` method for consistency